src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1993,4 +1980,30 @@ private function isFromTrustedProxy()
     {
         return self::$trustedProxies && IpUtils::checkIp($this->server->get('REMOTE_ADDR'), self::$trustedProxies);
     }
+
+    /**
+     * @param IpRetrieverInterface $ipRetriever
+     *
+     * @deprecated since version 2.7, to be removed in 3.0.
```

Comment:
It looks weird to introduce a new method and deprecate it at the same time. If any, you could just mark it as internal.